alf/trainers/policy_trainer.py

```diff
@@ -506,8 +510,13 @@ def play(root_dir,
             episode_reward = 0.
             episode_length = 0.
             episodes += 1
+            # observe the last step
+            for m in metrics:
```

Comment:
I think the metrics are already recorded by line 508-509? Is there any reason to record again using metrics?

Reply:
If you read the code carefully, you'll see, the last steps are not actually observed by the metrics above..

Reply:
Line 516 overwrites the last time_step of all episodes before it is observed by metrics.

Reply:
It's the first time_step (FIRST) not observed. Time steps >=1 till the LAST will always be counted by line 498 and 505. And for the initial time step it's probably fine because the reward dones't mean anything

Reply:
If you look at _step, it actually observes the input timestep, not the output timestep..

Reply:
I mean line 508-509

Reply:
Oh, that one is reported for every episode, and the final number is grand average of all episodes.  It's a useful thing to report both?

Reply:
OK fair enough